rem/remApp.swift

```diff
@@ -349,16 +351,36 @@ func drawStatusBarIcon(rect: CGRect) -> Bool {
     private func scheduleScreenshot(shareableContent: SCShareableContent) {
         Task {
             do {
-                guard isCapturing == .recording else { return }
+                guard isCapturing == .recording else { 
+                    logger.debug("Stopped Recording")
+                    return }
                 
                 var displayID: CGDirectDisplayID? = nil
-                if let screenID = NSScreen.main?.deviceDescription[NSDeviceDescriptionKey("NSScreenNumber")] as? NSNumber {
-                    displayID = CGDirectDisplayID(screenID.uint32Value)
-                    logger.debug("Display ID: \(displayID ?? 999)")
+                if settingsManager.settings.recordWindowWithMouse {
+                    let mouseLocation = NSEvent.mouseLocation
+                    if let screen = NSScreen.screens.first(where: {$0.frame.contains(mouseLocation)}) {
+                        if let screenID = screen.deviceDescription[NSDeviceDescriptionKey("NSScreenNumber")] as? NSNumber {
+                            displayID = CGDirectDisplayID(screenID.uint32Value)
```

Comment:
Put this first since it seems like this sometimes results a nil displayID when computer wakes up